### DISCUSSION TITLE
[question] 为什么electron的主进程和渲染进行使用同一个node_modules?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [alex8088/electron-vite#605] [Q&A]
请问一下， 为什么electron的主进程和渲染进行使用同一个node_modules呢?
主进程和渲染进程一个是后端一个是前端，为何都使用同一个node_modules，而不分开呢？

**Comment by takegine** [comment]
这有啥好分开的。难道存在某个依赖你在node和v8中要使用不同的版本？当然，如果你使用pnpm的话，你完全可以用 workspace来实现，用monorepo的逻辑，把main和renderer作为两个项目。那么他们会分别有package.json， 也会分别有node_modules
